op-conductor/consensus/raft.go

```diff
@@ -216,14 +218,21 @@ func (rc *RaftConsensus) CommitUnsafePayload(payload *eth.ExecutionPayloadEnvelo
 	if err := f.Error(); err != nil {
 		return errors.Wrap(err, "failed to apply payload envelope")
 	}
+	rc.log.Debug("unsafe payload committed", "number", uint64(payload.ExecutionPayload.BlockNumber), "hash", payload.ExecutionPayload.BlockHash.Hex())
+
+	// raft.Apply only guarantees that log entries are committed to majority of the followers, but it does not guarantee that the log entry is applied to the FSM.
+	// Utilize barrier here to make sure they're applied to the FSM.
+	if err := rc.r.Barrier(defaultTimeout).Error(); err != nil {
```

Comment:
Raft Barrier comment:
```
// Barrier is used to issue a command that blocks until all preceding
// operations have been applied to the FSM. It can be used to ensure the
// FSM reflects all queued writes. An optional timeout can be provided to
// limit the amount of time we wait for the command to be started. This
// must be run on the leader, or it will fail.
```

This is needed for one race condition situation where:
1. Seq1 (Leader) applied the unsafe head changes to the network
2. Apply successful and sequencer p2p block X out
3. Seq1 loses leadership right after p2p
4. Seq2 gains leadership and tries to start sequencing by comparing local unsafe head with the unsafe head within raft FSM
5. At this moment, without Barrier, the block within FSM could be block X or X-1
6. And if the p2p block X hasn't arrived at Seq2 yet, then Seq2 could start sequencing from block X-1, thus resulting in a 1 block reorg of unsafe